assets/content/cookbook/Beginner/arrays.md

```diff
@@ -77,152 +86,166 @@ That means these are your options:
 * If you iterate by increment the index you can get away with not increment the index it in case of removal (using `while` loop).
 * Or sometimes you might prefer to just build a new array instead.
 
-### Retrieving elements
+# Retrieving elements
 
 Array elements can be accessed through array notation, using the index of the element in the array.
 
 ```haxe
-var arrayOfStrings:Array<String> = ["first", "foo", "middle", "foo", "last"];
+var strings:Array<String> = ["first", "foo", "middle", "foo", "last"];
+
 // Retrieves first array element
-var first = arrayOfStrings[0];
+var first = strings[0];
+
 // Retrieves last array element
-var last = arrayOfStrings[arrayOfStrings.length - 1];
+var last = strings[strings.length - 1];
+
 // Retrieves first occurrence of "foo" string
-var firstOccurrence = arrayOfStrings[arrayOfStrings.indexOf("foo")];
+var first = strings[strings.indexOf("foo")];
+
 // Retrieves last occurrence of "foo" string
-var lastOccurrence = arrayOfStrings[arrayOfStrings.lastIndexOf("foo")];
+var last = strings[strings.lastIndexOf("foo")];
 ```
 
-### Iteration
+# Iteration
 
 The array defines an iterator, and its elements can therefore be iterated over.
 
 ```haxe
-for (item in arrayOfStrings) {
-    // do something
+for (item in items) {
+    trace(item);
 }
 ```
 
 You can also iterate using the array index:
 
 ```haxe
-var a1 = ["a", "b", "c"];
-var a2 = [11, 22, 33];
-for (i in 0...a1.length) {
-    trace(i + ", " + a1[i] + ", " + a2[i]);
+var items = ["a", "b", "c"];
+for (index in 0...items.length) {
+    trace('$index : ${items[index]}');
 }
 ```
 
-### Operations
+As an alternative, you can use the `keyValueIterator()` method to get indexes and values:
+
+```haxe
+var var items = ["a", "b", "c"];
+for (index => item in items.keyValueIterator()) {
```

Comment:
You don't need to (and shouldn't) call `keyValueIterator()` explicitly. Simply use `for (index => item in items)`.

Reply:
Ah, thanks - didn't know that!
Updated the example.